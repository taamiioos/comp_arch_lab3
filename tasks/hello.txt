.data
    str STRING hello world

.code
    ; загрузить адрес строки str в регистр r1
    LOAD r1, str

print:
    ; загрузить символ из текущей ячейки памяти в регистр r2
    LOAD r2, [r1]

    ; вывести символ на экран
    OUTPUT r2

    ; проверить, достигнут ли конец строки
    CMP r2, 0
    JZ print_end

    ; увеличить указатель на следующую ячейку памяти
    INC r1
    JMP print_string

end:
    ; остановить выполнение программы
    HLT
