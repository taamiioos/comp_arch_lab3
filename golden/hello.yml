in_source: |-
  .data
      STRING str "hello world"

  .code
      MOV r1 str
      LOAD_R r2 r1

  print_str:
      JZ_R end r2
      OUTPUT r2
      NEXT r2 r1
      JMP print_str

  end:
      HLT

in_stdin: |
out_stdout: |
  LoC: 15 Instr: 22
  ============================================================
  hello world
  Instructions: 48 Ticks: 93
out_code: |-
  [14,
   {"index": 0, "opcode": "number", "args": [1234]},
   {"index": 1, "opcode": "number", "args": [1234]},
   {"index": 2, "opcode": "string", "args": [104]},
   {"index": 3, "opcode": "string", "args": [101]},
   {"index": 4, "opcode": "string", "args": [108]},
   {"index": 5, "opcode": "string", "args": [108]},
   {"index": 6, "opcode": "string", "args": [111]},
   {"index": 7, "opcode": "string", "args": [32]},
   {"index": 8, "opcode": "string", "args": [119]},
   {"index": 9, "opcode": "string", "args": [111]},
   {"index": 10, "opcode": "string", "args": [114]},
   {"index": 11, "opcode": "string", "args": [108]},
   {"index": 12, "opcode": "string", "args": [100]},
   {"index": 13, "opcode": "string", "args": [0]},
   {"index": 0, "opcode": "mov", "args": [0, 2]},
   {"index": 1, "opcode": "load_r", "args": [1, 0]},
   {"index": 2, "opcode": "jz_r", "args": [6, 1]},
   {"index": 3, "opcode": "store_v", "args": [1, 1]},
   {"index": 4, "opcode": "next", "args": [1, 0]},
   {"index": 5, "opcode": "jmp", "args": [2]},
   {"index": 6, "opcode": "hlt", "args": []}]
out_log: |
  INFO     root:simulation.py:60 Start simulation
  DEBUG    root:simulation.py:26 TICK:    1 [ 1: LOAD_R ] PC:  1 DR:  0 SB:  0 RG: [2, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:    3 [ 2: JZ_R   ] PC:  2 DR:  2 SB:  0 RG: [2, 104, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:    5 [ 3: STORE_V] PC:  3 DR:  2 SB:  0 RG: [2, 104, 0, 0] 
  DEBUG    root:data_path.py:153 output: h << 104
  DEBUG    root:simulation.py:26 TICK:    7 [ 4: NEXT   ] PC:  4 DR:  1 SB:  0 RG: [2, 104, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   10 [ 5: JMP    ] PC:  5 DR:  3 SB:  0 RG: [3, 101, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   11 [ 2: JZ_R   ] PC:  2 DR:  3 SB:  0 RG: [3, 101, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   13 [ 3: STORE_V] PC:  3 DR:  3 SB:  0 RG: [3, 101, 0, 0] 
  DEBUG    root:data_path.py:153 output: he << 101
  DEBUG    root:simulation.py:26 TICK:   15 [ 4: NEXT   ] PC:  4 DR:  1 SB:  0 RG: [3, 101, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   18 [ 5: JMP    ] PC:  5 DR:  4 SB:  0 RG: [4, 108, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   19 [ 2: JZ_R   ] PC:  2 DR:  4 SB:  0 RG: [4, 108, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   21 [ 3: STORE_V] PC:  3 DR:  4 SB:  0 RG: [4, 108, 0, 0] 
  DEBUG    root:data_path.py:153 output: hel << 108
  DEBUG    root:simulation.py:26 TICK:   23 [ 4: NEXT   ] PC:  4 DR:  1 SB:  0 RG: [4, 108, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   26 [ 5: JMP    ] PC:  5 DR:  5 SB:  0 RG: [5, 108, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   27 [ 2: JZ_R   ] PC:  2 DR:  5 SB:  0 RG: [5, 108, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   29 [ 3: STORE_V] PC:  3 DR:  5 SB:  0 RG: [5, 108, 0, 0] 
  DEBUG    root:data_path.py:153 output: hell << 108
  DEBUG    root:simulation.py:26 TICK:   31 [ 4: NEXT   ] PC:  4 DR:  1 SB:  0 RG: [5, 108, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   34 [ 5: JMP    ] PC:  5 DR:  6 SB:  0 RG: [6, 111, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   35 [ 2: JZ_R   ] PC:  2 DR:  6 SB:  0 RG: [6, 111, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   37 [ 3: STORE_V] PC:  3 DR:  6 SB:  0 RG: [6, 111, 0, 0] 
  DEBUG    root:data_path.py:153 output: hello << 111
  DEBUG    root:simulation.py:26 TICK:   39 [ 4: NEXT   ] PC:  4 DR:  1 SB:  0 RG: [6, 111, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   42 [ 5: JMP    ] PC:  5 DR:  7 SB:  0 RG: [7, 32, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   43 [ 2: JZ_R   ] PC:  2 DR:  7 SB:  0 RG: [7, 32, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   45 [ 3: STORE_V] PC:  3 DR:  7 SB:  0 RG: [7, 32, 0, 0] 
  DEBUG    root:data_path.py:153 output: hello  << 32
  DEBUG    root:simulation.py:26 TICK:   47 [ 4: NEXT   ] PC:  4 DR:  1 SB:  0 RG: [7, 32, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   50 [ 5: JMP    ] PC:  5 DR:  8 SB:  0 RG: [8, 119, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   51 [ 2: JZ_R   ] PC:  2 DR:  8 SB:  0 RG: [8, 119, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   53 [ 3: STORE_V] PC:  3 DR:  8 SB:  0 RG: [8, 119, 0, 0] 
  DEBUG    root:data_path.py:153 output: hello w << 119
  DEBUG    root:simulation.py:26 TICK:   55 [ 4: NEXT   ] PC:  4 DR:  1 SB:  0 RG: [8, 119, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   58 [ 5: JMP    ] PC:  5 DR:  9 SB:  0 RG: [9, 111, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   59 [ 2: JZ_R   ] PC:  2 DR:  9 SB:  0 RG: [9, 111, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   61 [ 3: STORE_V] PC:  3 DR:  9 SB:  0 RG: [9, 111, 0, 0] 
  DEBUG    root:data_path.py:153 output: hello wo << 111
  DEBUG    root:simulation.py:26 TICK:   63 [ 4: NEXT   ] PC:  4 DR:  1 SB:  0 RG: [9, 111, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   66 [ 5: JMP    ] PC:  5 DR: 10 SB:  0 RG: [10, 114, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   67 [ 2: JZ_R   ] PC:  2 DR: 10 SB:  0 RG: [10, 114, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   69 [ 3: STORE_V] PC:  3 DR: 10 SB:  0 RG: [10, 114, 0, 0] 
  DEBUG    root:data_path.py:153 output: hello wor << 114
  DEBUG    root:simulation.py:26 TICK:   71 [ 4: NEXT   ] PC:  4 DR:  1 SB:  0 RG: [10, 114, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   74 [ 5: JMP    ] PC:  5 DR: 11 SB:  0 RG: [11, 108, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   75 [ 2: JZ_R   ] PC:  2 DR: 11 SB:  0 RG: [11, 108, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   77 [ 3: STORE_V] PC:  3 DR: 11 SB:  0 RG: [11, 108, 0, 0] 
  DEBUG    root:data_path.py:153 output: hello worl << 108
  DEBUG    root:simulation.py:26 TICK:   79 [ 4: NEXT   ] PC:  4 DR:  1 SB:  0 RG: [11, 108, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   82 [ 5: JMP    ] PC:  5 DR: 12 SB:  0 RG: [12, 100, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   83 [ 2: JZ_R   ] PC:  2 DR: 12 SB:  0 RG: [12, 100, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   85 [ 3: STORE_V] PC:  3 DR: 12 SB:  0 RG: [12, 100, 0, 0] 
  DEBUG    root:data_path.py:153 output: hello world << 100
  DEBUG    root:simulation.py:26 TICK:   87 [ 4: NEXT   ] PC:  4 DR:  1 SB:  0 RG: [12, 100, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   90 [ 5: JMP    ] PC:  5 DR: 13 SB:  0 RG: [13, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   91 [ 2: JZ_R   ] PC:  2 DR: 13 SB:  0 RG: [13, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   93 [ 6: HLT    ] PC:  6 DR: 13 SB:  0 RG: [13, 0, 0, 0] 
  INFO     root:simulation.py:42 output_buffer: hello world
  INFO     root:simulation.py:68 End simulation
